client/src/components/forms/MailingList/index.tsx

```diff
@@ -85,49 +92,28 @@ function MailingListForm({ classes }: WithStyles<typeof styles>) {
       </div>
 
       <div className={classes.slider}>
-        <div
-          className={classes.slide}
-          style={{ transform: `translateX(${translateValue}%)` }}
-        >
-          <EmailSlide email={email} setEmail={setEmail} />
-        </div>
-        <div
-          className={classes.slide}
-          style={{ transform: `translateX(${translateValue}%)` }}
-        >
-          <FeedbackSlide
-            events={events}
-            setEvents={setEvents}
-            feedback={feedback}
-            setFeedback={setFeedback}
-          />
-        </div>
-        <div
-          className={classes.slide}
-          style={{ transform: `translateX(${translateValue}%)` }}
-        >
-          <SocialSlide />
-        </div>
-      </div>
-
-      <div
-        className={classes.buttonContainer}
-        style={!isFirstSlide(translateValue) ? { width: '100%' } : {}}
-      >
-        {!isFirstSlide(translateValue) && !isLastSlide(translateValue) && (
-          <Button onClick={goLeft} className={classes.button}>
-            <NavigateBeforeIcon />
-            Back
-          </Button>
-        )}
-        {!isLastSlide(translateValue) && (
-          <Button onClick={goRight} className={classes.button}>
-            {translateValue === -100 * (NUMBER_OF_SLIDES - 2)
-              ? 'Submit'
-              : 'Next'}
-            <NavigateNextIcon />
-          </Button>
-        )}
+        {slideArr.map((s, index) => (
+          <div
+            key={index}
+            className={classes.slide}
+            style={{ transform: `translateX(${translateValue}%)` }}
+          >
```

Comment:
I think a better way to reuse this `div` (if that's what you're going for) would be to define a small subcomponent since it's not immediately obvious what's going on here. Just a nitpick tho 😅

Reply:
A sub component for the slide or for the slider?
I'm down for a slider sub component
What do you think makes it hard to read?

Reply:
A slider component would be very fresh but I meant the slide here.
It 100% doesn't need to be a subcomponent... that's probably overkill but I find the map and matching the index to be less readable than plainly having 3 of the same `div` with the slides inside. The map makes me have to think a little bit and check to see where the array is defined and if it is ever updated to be sure of what's going on whereas if it were just laid out it's blatantly obvious.
I assume the reason was to reuse the containing `div`?

Reply:
Oh I see what you're saying! Yes it was to reuse the `div`
Yea I can switch it back. There really is no major need for an array.
Will switch back!